### PR TITLE
DO NOT MERGE: General Review

### PR DESCRIPTION
**Pull request by furkanarabaci**

### CODE REVIEW

File: .ui/library/lviPerson.cpx [hunk 1]
```diff
@@ -25,15 +23,63 @@
 			}
 		},
 		{
-			"className": ".sf-label .sf-listViewItem-simple-title",
+			"className": ".sf-flexLayout .sf-listViewItem-simple .sf-listViewItem-inactive .sf-listViewItem-active",
```
Comment (furkanarabaci) [inline .ui/library/lviPerson.cpx:26]: Both inactive and active classes are here. You might want to remove one of them.

File: .ui/library/lviPerson.cpx [hunk 2]
```diff
@@ -25,15 +23,63 @@
...
+				"parent": "1fef-4b87-213d-4951",
+				"usePageVariable": true
+			},
+			"source": {
+				"page": "__library__"
+			},
+			"type": "ImageView",
+			"userProps": {
+				"flexProps": {},
+				"height": 52.27272727272727,
```
Comment (furkanarabaci) [inline .ui/library/lviPerson.cpx:66]: On ImageView, it is better to specify the dimensions on a different class rather than user properties

File: .ui/library/lviProperty.cpx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+				"usePageVariable": true
+			}
+		},
+		{
+			"className": ".sf-textView",
+			"id": "77ca-dba6-8215-b42c",
+			"props": {
+				"children": [],
+				"html": "<span style=\"font-family: SFProText-Regular; font-size: 16px; white-space: pre-wrap; background-color: rgb(255, 255, 255);\">Lorem ipsum dolor sit amet, consectetur \n  adipiscing elit. <span style=\"color: rgb(0, 161, 241); font-size: 17.3913px; font-family: SFProText-Regular; background-color: rgb(255, 255, 255);\">Donec quis semper justo. In  \n  ultrices metus nec sodales consectetur.</span> \n  Aliquam erat volutpat. <span style=\"font-size: 16px; font-family: SFProText-Regular; background-color: rgb(255, 255, 255); text-decoration-color: rgb(0, 0, 0); text-decoration-line: underline;\">Suspendisse</span> potenti.</span><br>",
+				"name": "lblValue",
```
Comment (furkanarabaci) [inline .ui/library/lviProperty.cpx:67]: I think we should have changed the name to tvValue, since it is textview now

File: .ui/library/materialTextBox.cpx [hunk 4]
```diff
@@ -1,22 +0,0 @@
-{
-	"components": [
```
Comment (furkanarabaci) [inline .ui/library/materialTextBox.cpx:2]: This file was needed, you are gonna need it on login page 😆

File: scripts/app.ts [hunk 5]
```diff
@@ -11,10 +11,11 @@ import router from "routes";
 // trigger onUnhandledError callback.
 Application.onUnhandledError = function (e: UnhandledError) {
     const error = errorStackBySourceMap(e);
+    console.log(error);
```
Comment (furkanarabaci) [inline scripts/app.ts:14]: console.error instead of console.log 😉

File: scripts/components/LviPerson.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    constructor(props?: any, pageName?: string) {
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+    get name(): string {
+        return this.lblName.text || '';
+    }
+    set name(value: string) {
+        this.lblName.text = value;
```
Comment (furkanarabaci) [inline scripts/components/LviPerson.ts:15]: It is better to handle falsy values like `this.lblName.text = value || ''`

File: scripts/components/LviProperty.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    constructor(props?: any, pageName?: string) {
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+    get keyText(): string {
+        return this.lblKey.text || '';
+    }
+    set keyText(value: string) {
+        this.lblKey.text = value;
```
Comment (furkanarabaci) [inline scripts/components/LviProperty.ts:14]: It is better to handle falsy values like `this.lblValue.text = value || ''`

File: scripts/components/MaterialTextBox.ts [hunk 8]
```diff
@@ -1,3 +0,0 @@
-import MaterialTextBox from 'materialtextbox';
```
Comment (furkanarabaci) [inline scripts/components/MaterialTextBox.ts:1]: You are gonna need this file 😉

File: scripts/duck/people/reducer.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { Constants, ActionTypes, State } from ".";
+
+const initialState: State = {
+    peopleList: [],
```
Comment (furkanarabaci) [inline scripts/duck/people/reducer.ts:4]: The semicolon at the end might be dangerous. This will be interpreted as `initialState = { peopleList: [], undefined }`

File: scripts/lib/language-helper.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import Data from "sf-core/global/data";
+
```
Comment (furkanarabaci) [inline scripts/lib/language-helper.ts:2]: It is practically better to do something like `Data.getStringVariable('language') || Data.setStringVariable('language', 'en');` // or whatever your default language is to not worry about non existing cases

File: scripts/pages/pgPeopleDetail.ts [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+ */
+function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
+        const people: People = store.getState().people.peopleList[this.routeData.peopleIndex];
+        this.peopleName = people.name;
+        this.details = Object.entries(people).filter((value) => ['string', 'number', 'boolean'].includes(typeof value[1]))
+        this.showImage();
+        this.initHeader();
+        this.initListView();
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleDetail.ts:74]: This method should locate in onLoad. It does nothing but define callbacks, which doesn't change on runtime.

File: scripts/pages/pgPeopleDetail.ts [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+ * @event onShow
+ * This event is called when a page appears on the screen (everytime).
+ */
+function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
+        const people: People = store.getState().people.peopleList[this.routeData.peopleIndex];
+        this.peopleName = people.name;
+        this.details = Object.entries(people).filter((value) => ['string', 'number', 'boolean'].includes(typeof value[1]))
+        this.showImage();
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleDetail.ts:72]: This method doesn't need to be defined in onShow.

File: scripts/pages/pgPeopleDetail.ts [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    }
+}
+
+/**
+ * @event onShow
+ * This event is called when a page appears on the screen (everytime).
+ */
+function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleDetail.ts:68]: The data assignments under this condition can be moved to another function to increase readability

File: scripts/pages/pgPeopleLinkDetail.ts [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    initHeader() {
+        this.headerBar.title = this.key;
+    }
+    async fetchData() {
+        try {
+            const res = await execute(this.value);
+            this.data = Object.entries(res).filter((value) => ['string', 'number', 'boolean'].includes(typeof value[1]))
+        } catch (error) {
+            console.log(error)
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleLinkDetail.ts:24]: You might want to throw your error instead of logging it, so the `Application.onUnhandledError` method we defined in app.ts would trigger for better error handling.

File: scripts/pages/pgPeopleLinkDetail.ts [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {Object} parameters passed from Router.go function
+ */
+async function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
+        this.key = this.routeData.key;
+        this.value = this.routeData.value;
+        await this.fetchData();
+        this.initHeader();
+        this.initListView();
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleLinkDetail.ts:54]: This method should locate in onLoad. It does nothing but define callbacks, which doesn't change on runtime.

File: scripts/pages/pgPeopleLinkDetail.ts [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * @event onShow
+ * This event is called when a page appears on the screen (everytime).
+ * @param {function} superOnShow super onShow function
+ * @param {Object} parameters passed from Router.go function
+ */
+async function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleLinkDetail.ts:49]: The data assignments under this condition can be moved to another function to increase readability

File: scripts/pages/pgPeopleLinkDetail.ts [hunk 17]
```diff
@@ -0,0 +1,66 @@
+import PgPeopleLinkDetailDesign from 'generated/pages/pgPeopleLinkDetail';
+import LviProperty from 'components/LviProperty';
+import { execute } from 'services/people'
+
+export default class PgPeopleLinkDetail extends PgPeopleLinkDetailDesign {
+    key: string;
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleLinkDetail.ts:6]: There is a small risk of those variables being undefined. You might want to handle them with `key = ''`

File: scripts/pages/pgPeopleLinkDetail.ts [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+ * This event is called when a page appears on the screen (everytime).
+ * @param {function} superOnShow super onShow function
+ * @param {Object} parameters passed from Router.go function
+ */
+async function onShow(superOnShow: () => void) {
+    superOnShow();
+    if (this.routeData) {
+        this.key = this.routeData.key;
+        this.value = this.routeData.value;
+        await this.fetchData();
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleLinkDetail.ts:52]: I assume await wouldn't work here, since the function is a callback. You don't need to anyways

File: scripts/pages/pgPeopleList.ts [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+        this.onHide = onHide.bind(this, this.onHide && this.onHide.bind(this));
+    }
+    async fetchPeople() {
+        try {
+            this.page++;
+            await peopleService.getAll(this.page);
+            this.servicePage++;
+            this.refreshListView();
+        } catch (error) {
+            console.log('fetchPeopleError', error)
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleList.ts:35]: Do not log errors with console.log 😉  use throw

File: scripts/pages/pgPeopleList.ts [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+                detailItem.onPress = ({ index }) => {
+                    this.lvPerson.refreshRowRange({ itemCount: 1, positionStart: index });
+                    this.router.push("/pages/pgPeopleDetail", { peopleIndex: index });
+                };
+                this.applyDimension(detailItem);
+                return [detailItem];
+            }
+        };
+    }
+    applyDimension(item: any): void {
```
Comment (furkanarabaci) [inline scripts/pages/pgPeopleList.ts:96]: If you looked for a parent class for every UI component with child, you can use `import ViewGroup from 'sf-core/ui/viewgroup`. But it would be better to use ListViewItem

File: scripts/pages/pgSettings.ts [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    router: any;
+    currentTheme: string;
+    constructor() {
+        super();
+        // Overrides super.onShow method
+        this.onShow = onShow.bind(this, this.onShow.bind(this));
+        // Overrides super.onLoad method
+        this.onLoad = onLoad.bind(this, this.onLoad.bind(this));
+
+        this.currentTheme = Data.getStringVariable('currentTheme');
```
Comment (furkanarabaci) [inline scripts/pages/pgSettings.ts:21]: You could have wrapped this method inside of theme.ts for better readability. But this is also fine.

File: scripts/pages/pgSettings.ts [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+                            index: AlertView.ButtonType.NEUTRAL,
+                            text: SMF.i18n.languageKV[lang].restart,
+                            onClick: function (): void {
+                                Application.restart();
+                            }
+                        });
+                        alertView.show();
+                    }
+                });
+                menu.items.push(menuItem);
```
Comment (furkanarabaci) [inline scripts/pages/pgSettings.ts:65]: For iOS, a cancel button is missing. The users that try to close the menu, can't.

File: scripts/services/serviceConfig.ts [hunk 23]
```diff
@@ -0,0 +1,9 @@
+import ServiceCall from "sf-extension-utils/lib/service-call";
+export const sc = new ServiceCall({
+    baseUrl: "https://swapi.dev/api",
+    logEnabled: true,
+});
+export const scEmpty = new ServiceCall({
```
Comment (furkanarabaci) [inline scripts/services/serviceConfig.ts:6]: A weird way of doing things. Why did you need something like this ?

File: themes/baseTheme/styles/common.json [hunk 24]
```diff
@@ -1,2 +1,29 @@
 {
+	".sf": {
```
Comment (furkanarabaci) [inline themes/baseTheme/styles/common.json:2]: Could you move those classes out of common.json ? Could you do it like it's done on defaultTheme ?

**Review by furkanarabaci — COMMENTED** [review]
Please check... 🥀

File: .ui/library/lviPerson.cpx (hunk 1 above)
Comment (berkbaski) [reply .ui/library/lviPerson.cpx:26]: I think we don't need both! Because listView will render what need. I will remove both.

File: scripts/services/serviceConfig.ts (hunk 23 above)
Comment (berkbaski) [reply scripts/services/serviceConfig.ts:6]: Because I use two kinds of http-request.

1. Getting all data.
2. Getting specific data from some link fields of my first data.

I did request to `http://swapi.dev/api/people/` and I get the data. I do clickable URL some fields. 

### For ex: 
````json
{
   "name":"Luke Skywalker",
   "height":"172",
   "mass":"77",
   "hair_color":"blond",
   "skin_color":"fair",
   "eye_color":"blue",
   "birth_year":"19BBY",
   "gender":"male",
   "homeworld":"http://swapi.dev/api/planets/1/",
   "films":[
      "http://swapi.dev/api/films/1/",
      "http://swapi.dev/api/films/2/",
      "http://swapi.dev/api/films/3/",
      "http://swapi.dev/api/films/6/"
   ],
   "species":[
      
   ],
   "vehicles":[
      "http://swapi.dev/api/vehicles/14/",
      "http://swapi.dev/api/vehicles/30/"
   ],
   "starships":[
      "http://swapi.dev/api/starships/12/",
      "http://swapi.dev/api/starships/22/"
   ],
   "created":"2014-12-09T13:50:51.644000Z",
   "edited":"2014-12-20T21:17:56.891000Z",
   "url":"http://swapi.dev/api/people/1/"
}
````

There is my first model. I do clickable to `homeworld`, `url`  fields. But I dynamically render object values I mean I don't know what the object has which keys or values. Because I created one more `ServiceCall` object and someone has no baseURL. I use the `has empty baseURL ServiceCall` object if the row is link.

I hope it's clear.

Comment (furkanarabaci) [reply scripts/services/serviceConfig.ts:6]: You could have easily done that with `sc.request('/people/', { method: "GET" ))`. Creating another service-call instance is not necessary unless you are fetching data from a different hostname. That's the point of using a single instance 😉

Comment (furkanarabaci) [reply scripts/services/serviceConfig.ts:6]: I am getting you now. You have an unknown url and you don't want to deal with separating it. On larger scale projects, you would need to parse the URL and use the same instance, to avoid maintenance issues. For this case, it isn't needed.
